### PR TITLE
chore: remove useless aliases for DM Dev docs 

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added or deleted? (Required)

This PR removes useless aliases that start with '/zh/tidb-data-migration/dev/' for DM Dev docs. Those aliases are useless because all the existing DM Dev docs will be directed to https://docs.pingcap.com/zh/tidb/dev/dm-overview so users could be aware of the doc location changes easily. 

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [x] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/handle-failed-ddl-statements.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 title: 处理出错的 DDL 语句
 summary: 了解在使用 TiDB Data Migration 迁移数据时，如何处理出错的 DDL 语句。
-aliases: ['/docs-cn/tidb-data-migration/dev/skip-or-replace-abnormal-sql-statements/','/zh/tidb-data-migration/dev/skip-or-replace-abnormal-sql-statements','/zh/tidb-data-migration/dev/handle-failed-sql-statements']
+aliases: ['/docs-cn/tidb-data-migration/dev/skip-or-replace-abnormal-sql-statements/']
```
Comment (TomShawn) [inline dm/handle-failed-ddl-statements.md:4]: /zh/tidb-data-migration/dev/skip-or-replace-abnormal-sql-statements 就会是死链。看起来这篇文档在迁移前就改过一次文件名了

File: dm/maintain-dm-using-tiup.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 title: 使用 TiUP 运维 DM 集群
 summary: 学习如何使用 TiUP 运维 DM 集群。
-aliases: ['/docs-cn/tidb-data-migration/dev/cluster-operations/','/zh/tidb-data-migration/dev/cluster-operations']
+aliases: ['/docs-cn/tidb-data-migration/dev/cluster-operations/']
```
Comment (TomShawn) [inline dm/maintain-dm-using-tiup.md:4]: ditto

**Review by TomShawn — APPROVED** [review]

File: dm/handle-failed-ddl-statements.md (hunk 1 above)
Comment (qiancai) [reply dm/handle-failed-ddl-statements.md:4]: 是的，等都统一重定向到 https://docs.pingcap.com/zh/tidb/dev/dm-overview 后，至少还能打开。按照之前的方式，批量定向到 /zh/tidb/dev/skip-or-replace-abnormal-sql-statements 就会是 404

Comment (TomShawn) [reply dm/handle-failed-ddl-statements.md:4]: 好的。LGTM

**Review by Austaras — APPROVED** [review]

**Pull request merged**